spring-aot/src/main/java/org/springframework/cloud/function/FunctionTypeProcessor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.function;
+
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.springframework.aot.context.bootstrap.generator.infrastructure.nativex.BeanFactoryNativeConfigurationProcessor;
+import org.springframework.aot.context.bootstrap.generator.infrastructure.nativex.NativeConfigurationRegistry;
+import org.springframework.aot.support.BeanFactoryProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.cloud.function.context.catalog.FunctionTypeUtils;
+import org.springframework.nativex.hint.TypeAccess;
+
+/**
+ * Ensures that Function/Consumer input types declared b the user are reflectively available.
+ *
+ * @author Oleg Zhurakousky
+ *
+ */
+public class FunctionTypeProcessor implements BeanFactoryNativeConfigurationProcessor {
+
+	@Override
+	public void process(ConfigurableListableBeanFactory beanFactory,
+			NativeConfigurationRegistry registry) {
+		new Processor().process(beanFactory, registry);
+	}
+
+	private static class Processor {
+
+		void process(ConfigurableListableBeanFactory beanFactory, NativeConfigurationRegistry registry) {
+			final Set<String> added = new HashSet<>();
+			new BeanFactoryProcessor(beanFactory).processBeans(this::isFunction,
+				(beanName, functionBeanType) -> {
+					Type functionType = FunctionTypeUtils.discoverFunctionTypeFromClass(functionBeanType);
+					Type inputType = FunctionTypeUtils.getInputType(functionType);
+					String name = inputType.getTypeName();
+					if (!name.startsWith("java.") &&
+						!name.startsWith("javax.")) {
+						if (added.add(name)) {
+							registry.reflection().forType(FunctionTypeUtils.getRawType(inputType))
+								.withAccess(TypeAccess.DECLARED_CONSTRUCTORS, TypeAccess.PUBLIC_CONSTRUCTORS, TypeAccess.DECLARED_FIELDS, TypeAccess.PUBLIC_FIELDS, TypeAccess.DECLARED_METHODS, TypeAccess.PUBLIC_METHODS);
```

Comment:
Are `TypeAccess.DECLARED_FIELDS` and `TypeAccess.PUBLIC_FIELDS` really needed here?

Reply:
Probably not. It was mainly cut/paste from the 'alike' component, so i'll fix it